src/modules/kpi-reporting/lib/mappers.js

```diff
@@ -0,0 +1,53 @@
+'use-strict';
+
+const months = ['January', 'February', 'March', 'April', 'May', 'June',
+  'July', 'August', 'September', 'October', 'November', 'December'];
+
+const mapReturnsDataByCycle = (data, returnCycle) => {
+  return {
+    ...returnCycle,
+    ...data,
+    total: data.internalOnTime + data.internalLate + data.externalOnTime + data.externalLate
+  };
+};
+
+const mapReturnsDataMonthly = (data) => {
+  const year = new Date().getFullYear();
+  return data.reduce((acc, row) => {
+    acc.totals.allTime = acc.totals.allTime + row.return;
+    acc.totals.ytd = acc.totals.ytd + (row.currentYear ? row.return : 0);
+    if (row.currentYear) {
+      acc.monthly.push({ ...row, month: months[(row.month - 1)], currentYear: year });
+    }
+    return acc;
+  }, { totals: { allTime: 0, ytd: 0 }, monthly: [] });
+};
+
+const percentChange = (data, index, key) => {
```

Comment:
Could do with a comment that indicates what data might look like

Reply:
done